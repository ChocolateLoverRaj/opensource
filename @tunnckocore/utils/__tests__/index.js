'use strict';

const path = require('path');
const { isMonorepo, getWorkspacesAndExtensions, createAliases } = require('..');

test('get extensions and workspaces - no workspaces', () => {
  const rootDir = path.dirname(__dirname);
  const result = getWorkspacesAndExtensions(rootDir);

  expect(result.workspaces).toStrictEqual([]);
  expect(result.lerna).toStrictEqual({});
  expect(result.exts).toStrictEqual(['tsx', 'ts', 'jsx', 'js', 'mjs']);
  expect(result.extensions).toStrictEqual([
    '.tsx',
    '.ts',
    '.jsx',
    '.js',
    '.mjs',
  ]);
});

test('getWorkspacesAndExtensions - correct workspaces', () => {
  const rootDir = path.join(__dirname, 'fixtures', 'lerna');

  const result = getWorkspacesAndExtensions(rootDir);

  expect(result.workspaces).toStrictEqual(['@tunnckocore', '@helios']);
  expect(result.extensions).toStrictEqual(['.js', '.jsx', '.ts']);
  expect(result.exts).toStrictEqual(['js', 'jsx', 'ts']);
  expect(result.lerna).toStrictEqual({
    packages: ['@tunnckocore/*', '@helios/*'],
  });
  expect(result.pkg).toStrictEqual({
    name: 'lerna-monorepo',
    extensions: result.exts,
  });
});

test('createAliases return empty alias object', () => {
  const cwd = path.dirname(path.dirname(__dirname));
  const result = createAliases(cwd);

  expect(result.alias).toStrictEqual({});
});

test('createAliases return correct aliases for yarn workspaces', () => {
  const yarnRoot = path.join(__dirname, 'fixtures', 'yarn-workspaces');
  const result = createAliases(yarnRoot);

  // eslint-disable-next-line unicorn/consistent-function-scoping
  const toAliases = (src) =>
    ['@hela/fab', '@tunnckocore/qux'].reduce((acc, name) => {
      acc[name] = path.join(yarnRoot, name, src);
      return acc;
    }, {});

  expect(result.alias).toStrictEqual(toAliases('src'));
  expect(result.lerna).toStrictEqual({});
  expect(result.exts).toStrictEqual(['js', 'ts']);
  expect(result.pkg).toStrictEqual({
    extensions: result.exts,
    workspaces: ['@tunnckocore/*', '@hela/*'],
  });
  expect(result.lernaPath).toStrictEqual(path.join(yarnRoot, 'lerna.json'));
  expect(result.packagePath).toStrictEqual(path.join(yarnRoot, 'package.json'));
});

test('createAliases return correct aliases for Lerna workspaces', () => {
  const lernaRoot = path.join(__dirname, 'fixtures', 'lerna');
  const res = createAliases(lernaRoot);

  // eslint-disable-next-line unicorn/consistent-function-scoping
  const toAliases = (src) =>
    ['@helios/foo', '@tunnckocore/barry'].reduce((acc, name) => {
      acc[name] = path.join(lernaRoot, name, src);
      return acc;
    }, {});

  expect(res.alias).toStrictEqual(toAliases(''));
  expect(res.lerna).toStrictEqual({
    packages: ['@tunnckocore/*', '@helios/*'],
  });
  expect(res.exts).toStrictEqual(['js', 'jsx', 'ts']);
  expect(res.pkg).toStrictEqual({
    name: 'lerna-monorepo',
    extensions: ['js', 'jsx', 'ts'],
  });
  expect(res.lernaPath).toStrictEqual(path.join(lernaRoot, 'lerna.json'));
  expect(res.packagePath).toStrictEqual(path.join(lernaRoot, 'package.json'));
});

test('isMonorepo - true for workspaces root', () => {
  const lernaRepo = path.join(__dirname, 'fixtures', 'lerna');
  expect(isMonorepo(lernaRepo)).toStrictEqual(true);
});

test('isMonorepo - false regular repository', () => {
  expect(isMonorepo(path.dirname(__dirname))).toStrictEqual(false);
});
