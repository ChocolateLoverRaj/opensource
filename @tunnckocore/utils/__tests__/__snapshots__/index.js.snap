// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get extensions and workspaces - no workspaces 1`] = `
Object {
  "extensions": Array [
    ".tsx",
    ".ts",
    ".jsx",
    ".js",
    ".mjs",
  ],
  "exts": Array [
    "tsx",
    "ts",
    "jsx",
    "js",
    "mjs",
  ],
  "lerna": Object {},
  "lernaPath": "/home/charlike/github/tunnckoCore/opensource/@tunnckocore/utils/lerna.json",
  "packagePath": "/home/charlike/github/tunnckoCore/opensource/@tunnckocore/utils/package.json",
  "pkg": Object {
    "author": "Charlike Mike Reagent <opensource@tunnckocore.com>",
    "dependencies": Object {},
    "description": "Thin layer on top of \`execa\` that allows executing multiple commands in parallel or in sequence with control for concurrency",
    "devDependencies": Object {},
    "engines": Object {
      "node": ">=8.11",
    },
    "files": Array [
      "dist",
    ],
    "homepage": "https://github.com/tunnckoCore/opensource",
    "keywords": Array [
      "tunnckocorehq",
      "develop",
      "parallel",
      "series",
      "concurrency",
      "sequence",
      "exec",
      "child",
      "process",
      "execute",
      "fork",
      "execfile",
      "execa",
      "spawn",
      "file",
      "shell",
      "bin",
      "binary",
      "binaries",
      "npm",
      "path",
      "local",
    ],
    "license": "MPL-2.0",
    "licenseStart": 2019,
    "main": "src/index.js",
    "module": "src/index.js",
    "name": "@tunnckocore/utils",
    "peerDependencies": Object {},
    "publishConfig": Object {
      "access": "public",
      "tag": "latest",
    },
    "repository": Object {
      "directory": "@tunnckocore/utils",
      "type": "git",
      "url": "https://github.com/tunnckoCore/opensource.git",
    },
    "scripts": Object {},
    "types": "dist/types/index.d.ts",
    "version": "0.4.2",
  },
  "workspaces": Array [],
}
`;
